src/topotoolbox/stream_object.py

```diff
@@ -0,0 +1,78 @@
+"""This module contains the StreamObject class.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from .flow_object import FlowObject
+
+# pylint: disable=no-name-in-module
+from . import _flow  # type: ignore
+
+_all_ = ['StreamObject']
+
+
+class StreamObject():
+    """A class to represent stream flow accumulation based on a FlowObject.
+    """
+
+    def __init__(self, flow: FlowObject) -> None:
+
+        acc = np.zeros_like(flow.target, order='F', dtype=np.float32)
+        weights = np.ones_like(flow.target, order='F', dtype=np.float32)
+
+        _flow.flow_accumulation(
+            acc, flow.source, flow.direction, weights, flow.shape)
```

Comment:
`flow_accumulation` could be a method on `FlowObject` that returns a `GridObject`. It is useful on its own.

It might also make constructing a `StreamObject` simpler: the constructor really takes a `FlowObject` and a Boolean `GridObject` and extracts the stream network indices from the `GridObject` and copies the `source`, `target` and `direction` members from the `FlowObject`, which it needs to route flow along the stream network.

If the user passes a threshold value rather than a `GridObject` then we can use `flow_accumulation` followed by a comparison to construct a Boolean `GridObject` and then use the previous constructor.